File: src/net/eduard/api/server/minigame/MinigameLobby.java
package net.eduard.api.server.minigame;

import java.util.ArrayList;

import net.eduard.api.lib.storage.Storable;

public class MinigameLobby implements Storable{
	
	private int id=1;
	
	private int slot;
	

	private transient ArrayList<MinigamePlayer> players = new ArrayList<>();

	public ArrayList<MinigamePlayer> getPlayers() {
		return players;
	}
	
	public void join(MinigamePlayer player) {
		for (MinigamePlayer p : players) {
			p.show(player);
			player.show(p);
		}
		player.setLobby(this);
		if (!players.contains(player))
		players.add(player);
		
	}
	public void leave(MinigamePlayer player) {
		for (MinigamePlayer p : players) {
			p.hide(player);
			player.hide(p);
		}
		player.setLobby(null);
		players.remove(player);
	}
	public void setPlayers(ArrayList<MinigamePlayer> players) {
		this.players = players;
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public int getSlot() {
		return slot;
	}

	public void setSlot(int slot) {
		this.slot = slot;
	}

}

File: src/main/java/net/eduard/api/server/minigame/MinigameLobby.kt
package net.eduard.api.server.minigame


/**
 * Representa o lobby do minigame , precisa ser extendido
 */
open class MinigameLobby() {

    var id = 1
    var serverName = "lobby-01"
    var playersAmount = 0
    var playersLimit = 20
    @Transient
    var players = mutableListOf<MinigamePlayer>()




    fun join(player: MinigamePlayer) {
        for (loopPlayer in players) {
            loopPlayer.show(player)
            player.show(loopPlayer)
        }
        player.lobby = this
        if (!players.contains(player))
            players.add(player)
    }

    fun leave(gamePlayer: MinigamePlayer) {
        gamePlayer.lobby = null
        players.remove(gamePlayer)
        for (loopPlayer in players) {
            loopPlayer.hide(gamePlayer)
            gamePlayer.hide(loopPlayer)
        }
    }

}
